### PR TITLE
📠 Playbook template

### PR DESCRIPTION
**Pull request by jphastings**
Improves on #41 so that there a playbook template is created when you fail that check.

### CODE REVIEW

File: lib/roo_on_rails/checks/documentation/playbook_template.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+_FIXME: go through this document and fill out the sections marked FIXME._
+
+## Responsible humans
+
+As a last resort, in an emergency, the following people have experience working with this service and have claimed responsibility for the running of this service. **If you have executed the plays below** and there is still a clear and present SEV-1 or greater issue, use the links below to get in touch:
+
+- Responsible McPerson ([PagerDuty](https://deliveroo.pagerduty.com/users/FIXME), [Workplace](https://deliveroo.facebook.com/profile.php?id=FIXME))
+
+## Owned Models
```
Comment (gregbeech) [inline lib/roo_on_rails/checks/documentation/playbook_template.md]: What's an "owned model"? If you mean the domain entities it's responsible for, is that necessarily important on-call information?

File: lib/roo_on_rails/checks/documentation/playbook_template.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+- _FIXME_ - a model representing `FIXME`.
+
+## Failure modes
```
Comment (gregbeech) [inline lib/roo_on_rails/checks/documentation/playbook_template.md]: You're alternating between `Sentence case` and `Title Case` in the template. Shall we go with `Title Case` everywhere?

File: lib/roo_on_rails/checks/documentation/playbook_template.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+### FIXME: an ability/functional area of the service
```
Comment (gregbeech) [inline lib/roo_on_rails/checks/documentation/playbook_template.md]: I'd be inclined to make this the alert name so people can grep the page for the alert name when they get it and hit on the way to fix it fast.

File: lib/roo_on_rails/checks/documentation/playbook_template.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+This document details the failure modes of this service and mitigating steps that can be taken.
+
+_FIXME: go through this document and fill out the sections marked FIXME._
+
+## Responsible humans
+
+As a last resort, in an emergency, the following people have experience working with this service and have claimed responsibility for the running of this service. **If you have executed the plays below** and there is still a clear and present SEV-1 or greater issue, use the links below to get in touch:
+
+- Responsible McPerson ([PagerDuty](https://deliveroo.pagerduty.com/users/FIXME), [Workplace](https://deliveroo.facebook.com/profile.php?id=FIXME))
```
Comment (gregbeech) [inline lib/roo_on_rails/checks/documentation/playbook_template.md]: Slack link might be more useful than Workplace.

File: lib/roo_on_rails/checks/documentation/playbook_template.md (hunk 1 above)
Comment (jphastings) [inline lib/roo_on_rails/checks/documentation/playbook_template.md]: That's what it was intended to represent — though it's not critical, I find it useful orienting information: "this service is down, I should be aware of problems with these domain concepts throughout the federation."

File: lib/roo_on_rails/checks/documentation/playbook_template.md (hunk 2 above)
Comment (jphastings) [inline lib/roo_on_rails/checks/documentation/playbook_template.md]: Oops! I'm happy with either - Title Case works for me

File: lib/roo_on_rails/checks/documentation/playbook_template.md (hunk 3 above)
Comment (jphastings) [inline lib/roo_on_rails/checks/documentation/playbook_template.md]: :+1: I'll add something to illustrate that; though there will be some which don't map to an alert (eg. [this one from actor tracking](https://github.com/deliveroo/actor-tracking/blob/master/docs/playbook.md#-disabling-a-feature-flagexperiment-when-florence-is-unavailable))

File: spec/roo_on_rails/checks/sidekiq/sidekiq_spec.rb [hunk 5]
```diff
@@ -3,52 +3,69 @@
...
-            before { File.delete('Procfile') rescue nil }
-            after { File.delete('Procfile') rescue nil }
-
-            it_expects_check_to_fail
+        } }
 
-            context 'when fixing' do
-              let(:perform) { silence_stream(STDOUT) { subject.fix } }
+        context "if no Procfile exists" do
+          before { File.delete('Procfile') rescue nil }
```
Comment (jphastings) [inline spec/roo_on_rails/checks/sidekiq/sidekiq_spec.rb]: Ensure we're testing when there is _actually_ no Procfile

File: spec/roo_on_rails/checks/sidekiq/sidekiq_spec.rb [hunk 6]
```diff
@@ -3,52 +3,69 @@
...
 
-              let(:procfile_contents) do
-                File.read('Procfile')
-              end
+          it_expects_check_to_fail
+        end
 
-              it_expects_check_to_pass
+        context "if Procfile doesn't have worker line" do
```
Comment (jphastings) [inline spec/roo_on_rails/checks/sidekiq/sidekiq_spec.rb]: Include tests for the Procfile existing

File: spec/roo_on_rails/checks/sidekiq/sidekiq_spec.rb [hunk 7]
```diff
@@ -3,52 +3,69 @@
...
       end
 
       context 'when sidekiq gems are not installed' do
+        let(:gem_output) { '' }
+
         it_expects_check_to_fail
       end
     end
+
+    context 'when sidekiq is disabled' do
```
Comment (jphastings) [inline spec/roo_on_rails/checks/sidekiq/sidekiq_spec.rb]: Include tests for when sidekiq ENV variable is disabled.

File: spec/support/check.rb [hunk 8]
```diff
@@ -8,23 +8,23 @@ def self.included(by)
 
     by.let(:shell) { FakeShell.new }
     by.let(:context) { Hashie::Mash.new }
-    by.subject { described_class.new(shell: shell, context: context) }
+    by.let(:check) { described_class.new(shell: shell, context: context) }
```
Comment (jphastings) [inline spec/support/check.rb]: A named `subject` means we can reset the subject to something else in specs, eg. the following for testing the `#fix` method

```ruby
subject(:fix) { check.fix }
```

File: spec/support/check.rb [hunk 9]
```diff
@@ -8,23 +8,23 @@ def self.included(by)
...
 
-    by.let(:perform) { silence_stream(STDOUT) { subject.call } }
+    by.subject(:perform) { silence_stream(STDOUT) { check.call } }
   end
 
   module Dsl
     def it_expects_check_to_fail(message = nil)
-      it { expect { perform rescue nil }.not_to change { context } }
+      it { expect { subject rescue nil }.not_to change { context } }
```
Comment (jphastings) [inline spec/support/check.rb]: This allows us to use the `it_expects_check_to_fail` helper even when within a different `subject`.

File: lib/roo_on_rails/checks/documentation/playbook_template.md [hunk 10]
```diff
@@ -0,0 +1,29 @@
+# Playbook
+
+This document details the failure modes of this service and mitigating steps that can be taken.
+
```
Comment (danielcooper) [inline lib/roo_on_rails/checks/documentation/playbook_template.md]: Maybe a note about adding a TLDR for what your service does?

**Comment by pedrocunha** [comment]
❤️  thanks JP

**Pull request merged**